File: iosApp/iosApp/feature/profile/settings/SettingsViewState.swift
//
//  SettingsViewState.swift
//  iosApp
//
//  Created by Марк Шавловский on 14.04.2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import Foundation

struct SettingsViewState {
    let phoneNumber: String
    let selectedCityName: String
    let state: State

    enum State {
        case success
        case loading
        case error
    }
}

File: feature/profile/src/commonMain/kotlin/com/bunbeauty/profile/ui/screen/settings/SettingsViewState.kt
package com.bunbeauty.profile.ui.screen.settings

import androidx.compose.runtime.Immutable
import com.bunbeauty.core.base.BaseViewState
import com.bunbeauty.designsystem.model.CityUI
import kotlinx.collections.immutable.ImmutableList

data class SettingsViewState(
    val phoneNumber: String,
    val selectedCityName: String,
    val state: State,
    val logoutUI: LogoutBottomSheetUI,
    val cityListBottomSheetUI: CityListBottomSheetUI,
    val disableUserBottomSheetUI: DisableUserBottomSheetUI,
) : BaseViewState {
    @Immutable
    data class LogoutBottomSheetUI(
        val isShown: Boolean,
    )

    @Immutable
    data class DisableUserBottomSheetUI(
        val isShown: Boolean,
    )

    @Immutable
    data class CityListBottomSheetUI(
        val isShown: Boolean,
        val cityListUI: ImmutableList<CityUI>,
    )

    enum class State {
        Success,
        Loading,
        Error,
    }
}
